dcos/http.py

```diff
@@ -79,6 +87,47 @@ def request(method,
                 response.status_code,
                 response.text)
 
+    return response
+
+
+def request(method,
+            url,
+            is_success=_default_is_success,
+            timeout=None,
+            **kwargs):
+    """Sends an HTTP request. If the server responds with 'authentication
```

Comment:
Since client code might want to make an http request without prompting for username/password, we should probably make that configurable with a kwarg